### DISCUSSION TITLE
Is it possible to dynamically pass in acr_values to an IdP?

### DISCUSSION THREAD

**Discussion by monfresh** [keycloak/keycloak#35608] [Q&A] [answered]
Hello. We are thinking of using Keycloak to configure login.gov as an IdP and CSP for our application. Login.gov allows SPs to restrict which MFA options a user can use. For example, by setting the `acr_values` to `http://idmanagement.gov/ns/assurance/aal/2?phishing_resistant=true` in the [authentication request](https://developers.login.gov/oidc/authorization/), login.gov will only allow the user to sign in using phishing-resistant methods (such as face/touch unlock, security keys, and PIV/CAC cards).

However, we don't want to force all of our users to use phishing-resistant options. We only want to enforce that for a subset of users, based on their role/organization in our application. So, what we're trying to figure out is whether it's possible to make a second request to the IdP (step-up auth essentially) using the `acrValues` option as documented in the [Keycloak JS adapter](https://www.keycloak.org/securing-apps/javascript-adapter). 

Will those acr values be appended to the authorization URL configured in Keycloak for the login.gov IdP?

**Comment by embesozzi** [comment, marked as answer]
In the Keycloak [Workshop: Step-up MFA with Biometrics](https://github.com/embesozzi/keycloak-workshop-stepup-mfa-biometrics) and the related article [Keycloak Step-up Authentication for Web and API](https://embesozzi.medium.com/keycloak-step-up-authentication-for-web-and-api-3ef4c9f25d42), you can find some examples. For instance, in the demo, you can see the use of ACR values in [oidc-stepup-demobank-spa/js/index.js#L9](https://github.com/embesozzi/keycloak-workshop-stepup-mfa-biometrics/blob/main/oidc-stepup-demobank-spa/js/index.js#L9). Additionally, there is a step-up example in the context of both applications and APIs.
On the other hand, if you're looking for phishing-resistant solutions such as passkeys, you may find this article helpful:  [Keycloak Workshop for Step Up with MFA Biometrics Authentication (Passkeys) and Passwordless experience with Passkey autofill](https://medium.com/@embesozzi/keycloak-workshop-for-step-up-with-mfa-biometrics-authentication-passkeys-b7020ea9ae1b).
